### ISSUE TITLE
httplog symlink support

### ISSUE THREAD

**Issue by Erwane** [rsyslog/rsyslog#601]
Hi,

Can you add the support of this wonderful feature of httplog ?

-s SYMLINK
              Create  a symlink to the currently active logfile.  This symlink
              does not have to be in  the  same  directory  as  the  currently
              active  logfile,  and  %  expandoes do not expand in the symlink
              filename.

Freshports : https://www.freshports.org/sysutils/httplog/
Source : ftp://ftp.nuug.no/pub/anders/distfiles/httplog-2.1.tar.gz*

Thanks

**Comment by rgerhards** [comment]
sorry, but I don't understand what you want rsyslog to do. Can you elaborate a little more?

**Comment by Erwane** [comment]
i have the apache log file, with date : ACCESS_LOG.201511
and httplog maintain a link to this current file : 
access.log -> ACCESS_LOG.201511

So, i can tail or do operations/Analyses on access.log, every day or month, name is the same.
Very useful.

my httplog config (in apache.conf) :
CustomLog     "|/usr/local/sbin/httplog -g log -s /var/log/apache/access.log /var/log/apache/ACCESS_LOG.%Y%m"

**Comment by rgerhards** [comment]
I still don't see what you want rsyslog to do. Are you talking to the right project? If you mean rsyslog, can you please say something along the lines of "I want imfile to do...". Thx

**Comment by Erwane** [comment]
I want rsyslog to write logs to file with name "log.%Y-%m-%d" AND maintain a symlink to current log file with simple name like "current-log"

Yes, it's the right project :)

**Comment by rgerhards** [comment]
ah, ok, now it's clear. Note to potential contributors: omfile would need to be enhanced.

**Comment by Erwane** [comment]
thanks

**Comment by madduck** [comment]
Please also see: http://rsyslog-users.1305293.n2.nabble.com/Maintining-a-symlink-to-a-log-file-td7589913.html

tl;dr: rsyslog does not really have a concept of "the current log file". The proper way to maintain this symlink seems to be to tell cron to create the symlink at 00:00:01 each day.

**Comment by sveniu** [comment]
An implementation could be made generic by allowing a command hook to be run when an output file is opened or closed. Instead of a specialized feature that deals with symlinks, a user would then be free to write exactly the script they need, and have it called with the opened/closed file's full path as an argument, and probably also the event type in an option or argument.

A user script can then do exactly what's needed, such as changing symlinks on 'file open' events, or initiating compression on 'file close' events, without having to revert to cron jobs or logrotate.

For reference, Apache's rotatelogs program [does exactly this](https://bz.apache.org/bugzilla/show_bug.cgi?id=51285) since version 2.4.

**Comment by Erwane** [comment]
any news for that ?

thanks ;)

**Comment by Erwane** [comment]
Hi, i'm re-upping this issue because it would be a really nice feature.
And i want to help but i can't, I have no experience on C language :(
